pkg/ddl/split_region.go

```diff
@@ -107,20 +115,54 @@ func preSplitPhysicalTableByShardRowID(ctx context.Context, store kv.SplittableS
 		key := tablecodec.EncodeRecordKey(recordPrefix, kv.IntHandle(recordID))
 		splitTableKeys = append(splitTableKeys, key)
 	}
-	var err error
-	regionIDs, err := store.SplitRegions(ctx, splitTableKeys, scatter, &tbInfo.ID)
+	var (
+		err     error
+		scatter bool
+		// PD controls the scope of scatter region through `tableId`.
+		// `nil` indicates that PD will scatter region at cluster level.
+		tableID *int64
+	)
+	switch scatterScope {
+	case scatterTable:
+		scatter = true
+		tableID = &tbInfo.ID
+	case scatterGlobal:
+		scatter = true
+		tableID = nil
+	default:
+		scatter = false
+		tableID = &tbInfo.ID
+	}
+	regionIDs, err := store.SplitRegions(ctx, splitTableKeys, scatter, tableID)
 	if err != nil {
 		logutil.DDLLogger().Warn("pre split some table regions failed",
 			zap.Stringer("table", tbInfo.Name), zap.Int("successful region count", len(regionIDs)), zap.Error(err))
 	}
-	regionIDs = append(regionIDs, splitIndexRegion(store, tbInfo, scatter)...)
+	regionIDs = append(regionIDs, splitIndexRegion(store, tbInfo, scatter, tableID)...)
 	return regionIDs
 }
 
 // SplitRecordRegion is to split region in store by table prefix.
-func SplitRecordRegion(ctx context.Context, store kv.SplittableStore, physicalTableID, tableID int64, scatter bool) uint64 {
+func SplitRecordRegion(ctx context.Context, store kv.SplittableStore, physicalTableID, tableID int64, scope scatterScope) uint64 {
 	tableStartKey := tablecodec.GenTablePrefix(physicalTableID)
-	regionIDs, err := store.SplitRegions(ctx, [][]byte{tableStartKey}, scatter, &tableID)
+	var (
```

Comment:
Can we merge line 148 ~ 164 into one funcs? since same code segments in multi place.